ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/TestDBLockStorageBackend.scala

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.ha
+
+import java.sql.{
+  Blob,
+  CallableStatement,
+  Clob,
+  Connection,
+  DatabaseMetaData,
+  NClob,
+  PreparedStatement,
+  SQLWarning,
+  SQLXML,
+  Savepoint,
+  Statement,
+  Struct,
+}
+import java.util.Properties
+import java.util.concurrent.Executor
+import java.{sql, util}
+
+import com.daml.platform.store.backend.DBLockStorageBackend
+
+class TestDBLockStorageBackend extends DBLockStorageBackend {
+
+  private var locks: Map[DBLockStorageBackend.Lock, Set[Connection]] = Map.empty
```

Comment:
This probably means the methods of this class are not safe to use concurrently. Could this be an issue? Could you document this at the class level?

Reply:
I think it should be thread safe, every place we modify/read this are in synchronized blocks

Reply:
You are right, I have missed the `synchronized`.

Reply:
Actually, I'm not sure. You can still call `tryAcquire` and `release` at the same time, right? And they both access the `locks` mutable value.